src/metal.cs

```diff
@@ -378,6 +437,32 @@ public partial interface MTLBlitCommandEncoder : MTLCommandEncoder {
 
 		[Abstract, Export ("copyFromBuffer:sourceOffset:toBuffer:destinationOffset:size:")]
 		void CopyFromBuffer (IMTLBuffer sourceBuffer, nuint sourceOffset, IMTLBuffer destinationBuffer, nuint destinationOffset, nuint size);
+		
+		[iOS (10,0), TV (10,0), NoWatch, NoMac]
+		[Abstract]
+		[Export ("updateFence:")]
+		void UpdateFence (IMTLFence fence);
+
+		[iOS (10,0), TV (10,0), NoWatch, NoMac]
+		[Abstract]
+		[Export ("waitForFence:")]
+		void WaitForFence (IMTLFence fence);
+	}
+	
+	public interface IMTLFence {}
+
+	[iOS (10,0), TV (10,0), NoWatch, NoMac]
+	[Protocol, Model]
```

Comment:
Also, specific to Metal (see comments in this file) several protocols are not meant to be user-created (but are exposed by objects Metal gives you) so a concrete type is only confusing / extra metadata.

> // From Apple Docs: Your app does not define classes that implement this protocol. Model is not needed